cmd/gsutil/main.go

```diff
@@ -58,12 +64,27 @@ func main() {
 	flag.Parse()
 	logger, _ := logging.NewLogger("", "gsutil")
 	defer logger.Sync()
-	gsutilPath, checkgsutilErr := exec.LookPath("gsutil")
-	if checkgsutilErr != nil {
-		logger.Fatalf("Error executing command 'gsutil verson'; err %s; output %s", checkgsutilErr.Error(), gsutilPath)
+
+	authFilePath := os.Getenv("GOOGLE_APPLICATION_CREDENTIALS")
+	if authFilePath != "" {
+		_, err := os.Stat(authFilePath)
+		if os.IsNotExist(err) {
+			logger.Fatalf("File does not exist at path %s; got err %s", authFilePath, err)
+		}
+
+		if _, checkgcloudErr := exec.LookPath("gcloud"); checkgcloudErr != nil {
+			logger.Fatalf("Error executing command 'gcloud'; err %s", checkgcloudErr)
+		}
+
+		if authCmdstdoutStderr, authErr := exec.Command("gcloud", "auth", "activate-service-account", "--key-file", authFilePath).CombinedOutput(); authErr != nil {
+			logger.Fatalf("Error executing command %s; cmd output: %s", authErr, authCmdstdoutStderr)
+		}
+		logger.Info("Successfully authenticated with gcloud")
 	}
 
-	logger.Infof("gsutil binary at path %s", gsutilPath)
+	if _, checkgsutilErr := exec.LookPath("gsutil"); checkgsutilErr != nil {
```

Comment:
hm, i find it kind of odd to use different variable names for `err` - although reusing `err` shadows previous usage, i feel like it's easier to read if we use `err` consistently

not a big deal though if you strongly prefer doing it this way!